### PR TITLE
Minor refactorings

### PR DESCRIPTION
**Pull request by MHaselmaier**
Hi,

I made some small refactorings.
In my opinion the added override specifiers are especially important.

### CODE REVIEW

File: src/oatpp/core/data/stream/BufferStream.cpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ oatpp::async::CoroutineStarter BufferOutputStream::flushToStreamAsync(const std:
       , m_stream(stream)
     {}
 
-    Action act() {
+    virtual Action act() override {
```
Comment (lganzzzo) [inline src/oatpp/core/data/stream/BufferStream.cpp:150]: Please use just `override`. No need in `virtual` here

File: src/oatpp/core/data/stream/Stream.cpp [hunk 2]
```diff
@@ -119,7 +119,7 @@ async::CoroutineStarter WriteCallback::writeExactSizeDataAsync(const void* data,
       , m_inlineData(data, size)
     {}
 
-    Action act() {
+    virtual Action act() override {
```
Comment (lganzzzo) [inline src/oatpp/core/data/stream/Stream.cpp:122]: Please use just `override`. No need in `virtual` here

File: src/oatpp/web/protocol/http/outgoing/Request.cpp [hunk 3]
```diff
@@ -165,7 +165,7 @@ oatpp::async::CoroutineStarter Request::sendAsync(std::shared_ptr<Request> _this
       , m_headersWriteBuffer(std::make_shared<oatpp::data::stream::BufferOutputStream>())
     {}
     
-    Action act() {
+    virtual Action act() override {
```
Comment (lganzzzo) [inline src/oatpp/web/protocol/http/outgoing/Request.cpp:168]: Please use just `override`. No need in `virtual` here

**Review by lganzzzo — APPROVED** [review]
Looks good!
Please address the minor comments.

**Comment by MHaselmaier** [comment]
I used both virtual and override because it was use in this way on other classes.
Would you be interessted in a global change to only use one of the virtual, override and final specifiers?
I could post another pull request, it should be an easy task with tool assistence.

**Comment by lganzzzo** [comment]
> Would you be interessted in a global change to only use one of the virtual, override and final specifiers?

Sounds good - we really need some code cleanup.
Also, please do not put final specifiers, since we want most of the APIs to be overridable.

**Comment by lganzzzo** [comment]
Merged!

**Pull request merged**